src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -230,8 +243,38 @@ public void RunClassInitialize(TestContext testContext)
                 throw new NullReferenceException(Resource.TestContextIsNull);
             }
 
-            // If class initialization is not done, then do it.
-            if (!this.IsClassInitializeExecuted)
+            MethodInfo initializeMethod = null;
+            string failedClassInitializeMethodName = string.Empty;
+
+            try
+            {
+                // If class initialization is not done, we can proceed to perform
+                // base class initialization first
+                if (!this.IsClassInitializeExecuted)
+                {
+                    // ClassInitialize methods for base classes are called in reverse order of discovery
+                    // Base -> Child TestClass
+                    var baseClassInitializeStack = new Stack<Tuple<MethodInfo, MethodInfo>>(this.BaseClassInitAndCleanupMethods);
+                    while (baseClassInitializeStack.Count > 0)
+                    {
+                        initializeMethod = baseClassInitializeStack.Pop().Item1;
+                        if (initializeMethod != null)
+                        {
+                            initializeMethod?.InvokeAsSynchronousTask(null, testContext);
```

Comment:
We do not need this null check operator here given that we are already performing a null check above.

Reply:
actually, i think we might needed it since the null check above is to make sure there are any non-null init method in the tuple that can be executed. In any case, I changed this for a `.Where` call when newing up the `baseClassInitializeStack `(on the `this.BaseClassInitAndCleanupMethods`) that way, we just pass executable init methods, and we don't need to do any extra thing to pass its corresponding cleanup method to the stack for cleanup methods to execute in the `RunClassCleanup` method.

Reply:
It looks like the updated code wouldn't run Cleanups from base classes that do not have an initialize method...